kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -112,7 +112,7 @@ termEquals
     -> TermLike variable
     -> MaybeT simplifier (OrPredicate variable)
 termEquals first second = MaybeT $ do
-    maybeResults <- BranchT.gather $ runMaybeT $ termEqualsAnd first second
+    maybeResults <- BranchT.gather $ runMaybeT $ termEqualsAnd Predicate.top first second
```

Comment:
Line lengths over 80 characters are discouraged by the style guide:
https://github.com/kframework/kore/wiki/Coding-Standards
We go over that limit when it significantly helps readability, e.g. in some tests and in the `andEqualsFunctions` implementation below.